libsolidity/formal/CHC.cpp

```diff
@@ -1633,27 +1661,26 @@ void CHC::checkAndReportTarget(
 	else if (result == CheckResult::SATISFIABLE)
 	{
 		solAssert(!_satMsg.empty(), "");
-		m_unsafeTargets[_target.errorNode].insert(_target.type);
 		auto cex = generateCounterexample(model, error().name);
 		if (cex)
-			m_errorReporter.warning(
+			m_unsafeTargets[_target.errorNode][_target.type] = {
 				_errorReporterId,
 				location,
 				"CHC: " + _satMsg + "\nCounterexample:\n" + *cex
-			);
+			};
```

Comment:
Why don't we just have
```cpp
if (m_settings.showUnproved)
 m_errorReporter.warning(...)
```
just like in the previous cases?

Reply:
Because there might be a lot of duplicates when multiple queries regarding the same target are unproved, or some are unproved and others are proved, in the CHC case. Storing them and deciding later what to report helps to only report one warning per target.